### DISCUSSION TITLE
Cannot generate an ERC20 token transfer transaction signature

### DISCUSSION THREAD

**Discussion by cyje66** [CoolBitX-Technology/coolwallet-signing-tools#54] [Q&A] [answered]
**Description**
In script-tester, testing `getERC20Script`in EthScript and get an error: `returnCode: 6AE2, message: called the script command to throw out the error`
script: `03000601C707000000003CCC07C002F800C2ACD70034FFF8C2A5D700FFF6C2ACD7002AFFF6CC071094CAAC270047CC07C01380B844a9059cbb000000000000000000000000CAA02700CC07200000000000000000000000000000000000000000CAA2C7000CCC0E1001C2E09700CC07C0028080C37097C002DC07C00345544811ACC7CC3E1D045B1507C004CC0F104012AC17C03F0401071507C002FF00B5AC17003FCAA6BF00DEF09700250F00CC0FC0023078BAA02F6C0E04DDF0970012AC17C03E0400141507C002FF00B5AC17003EDAA2C7B00CD207CC05065052455353425554546F4E`
argument: `15328000002c8000003c80000000000000000000000082571ED172c5E202b5309A94aEd0168Bc66F23A70000002386f26fc100000000000000003b9aca0c0000000000000000520800000000000000000001060455534454000000dac17f958d2ee523a2206206994597c13d831ec73045022009DE62567071D16F3104448C3553B908BEB9C6806F6C1C38C801EE7FDBD1523F022100E2CA0CAF5F59F64CDF91C2AD3B7AAAF3490224C41810AED666DE59623FB325B2`

The `argTokenInfo` and  `argSign` in the argument above is USDT described in [tokenType.ts](https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/master/packages/coin-eth/src/config/tokenType.ts).

SE version: 320

**Expected Behavior**
The `getETHScript `and `getEtherContractBlindScript` works fine and generates signatures, but the `getERC20Script` couldn't work.

**Screenshots or appendix**
![image](https://user-images.githubusercontent.com/23631550/154005775-e26fa22c-24b1-4e2f-ae75-dd5065c70fd2.png)

**Comment by hank121314** [comment, marked as answer]
The argument is incorrect.
You should check this function https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/cb985746ed746a56527819366de80c8c053f0b6a/packages/coin-eth/src/utils/scriptUtils.ts#L32-L53 to get the correct argument.
